### PR TITLE
[MenuItem] Pass popoverProps to menuItem

### LINKED ISSUE #1567: [Core] Menus - pass popoverProps property
I want to open submenu popover on click, not on hover. Seems like there is no way to control `interactionKind` inside menuItem component.

What about passing `popoverProps` object to menuItem, like in `Labs/Select`?

### PR DESCRIPTION
**Pull request by AlexMarvelo**
#### Fixes #1567 

Pass `popoverProps` object to `menuItem`, like in `Labs/Select`.
I'm not sure if there are any test needed. If yes, tell me

### CODE REVIEW

File: packages/core/src/components/menu/menuItem.tsx [hunk 1]
```diff
@@ -126,21 +130,25 @@ export class MenuItem extends AbstractComponent<IMenuItemProps, IMenuItemState>
...
                     enforceFocus={false}
                     hoverCloseDelay={0}
                     inline={true}
                     interactionKind={PopoverInteractionKind.HOVER}
                     position={this.state.alignLeft ? Position.LEFT_TOP : Position.RIGHT_TOP}
-                    popoverClassName={classNames(Classes.MINIMAL, Classes.MENU_SUBMENU, popoverClasses)}
                     useSmartArrowPositioning={false}
+                    {...popoverProps}
+                    content={submenuElement}
+                    popoverClassName={popoverClasses}
```
Comment (AlexMarvelo) [inline packages/core/src/components/menu/menuItem.tsx]: `content` can't be overwritten. `popoverProps.popoverClassName ` is added to `popoverClassName `

File: packages/core/src/components/menu/menuItem.tsx [hunk 2]
```diff
@@ -126,21 +130,25 @@ export class MenuItem extends AbstractComponent<IMenuItemProps, IMenuItemState>
...
             const submenuElement = <Menu ref={measureSubmenu}>{this.renderChildren()}</Menu>;
-            const popoverClasses = classNames({
-                [Classes.ALIGN_LEFT]: this.state.alignLeft,
-            });
+            const popoverClasses = classNames(
+                Classes.MINIMAL,
+                Classes.MENU_SUBMENU,
+                popoverProps.popoverClassName,
+                { [Classes.ALIGN_LEFT]: this.state.alignLeft },
+            );
```
Comment (giladgray) [inline packages/core/src/components/menu/menuItem.tsx]: nice! good refactor here

File: packages/core/src/components/menu/menuItem.tsx (hunk 1 above)
Comment (giladgray) [reply packages/core/src/components/menu/menuItem.tsx]: 💯

File: packages/core/test/menu/menuTests.tsx [hunk 3]
```diff
@@ -96,6 +96,26 @@ describe("MenuItem", () => {
...
+        const wrapper = shallow(
+            <MenuItem iconName="style" text="Style" popoverProps={popoverProps}>
+                <MenuItem text="one" />
+                <MenuItem text="two" />
+            </MenuItem>,
+        );
+        assert.strictEqual(wrapper.find(Popover).prop("inline"), popoverProps.inline);
+        assert.strictEqual(wrapper.find(Popover).prop("interactionKind"), popoverProps.interactionKind);
+        assert.notStrictEqual(wrapper.find(Popover).prop("popoverClassName").indexOf(popoverProps.popoverClassName), 0);
+        assert.notStrictEqual(wrapper.find(Popover).prop("content"), popoverProps.content);
```
Comment (AlexMarvelo) [inline packages/core/test/menu/menuTests.tsx]: Check `inline`, `interactionKind` (some of overwriteable props), `content` (shouldn't be passed) and `popoverClassName` (custom class should me included)

**Comment by adidahiya** [comment]
Thanks @AlexMarvelo!

**Pull request merged**